### PR TITLE
feat(cb2-6390): start/end date validation

### PR DESCRIPTION
**Pull request by leifkemp-bjss**
## Create validation for Test Date to prevent End date being earlier than the Start date

Test start and end dates are now compared to ensure end date is not behind start date
[CB2-6390](https://dvsa.atlassian.net/browse/CB2-6390)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

**Comment by Ruairidh-BJSS** [comment]
Missing tests

**Review by Ruairidh-BJSS — APPROVED** [review]

File: src/app/forms/validators/custom-validators.ts [hunk 1]
```diff
@@ -156,6 +156,25 @@ export class CustomValidators {
...
 
+  static aheadOfDate = (sibling: string): ValidatorFn => {
+    return (control: AbstractControl): ValidationErrors | null => {
+      if (control?.parent) {
+        const date = control.value;
+        console.log(date);
+
+        const siblingControl = control.parent.get(sibling) as CustomFormControl;
+        const siblingValue = siblingControl.value;
+        console.log(siblingValue);
```
Comment (Ruairidh-BJSS) [inline src/app/forms/validators/custom-validators.ts:167]: remove console logs pls

File: src/app/forms/templates/test-records/section-templates/test/specialist-test-section-group4.template.ts [hunk 2]
```diff
@@ -126,7 +126,11 @@ export const SpecialistTestSectionGroup4: FormNode = {
               label: 'Start time',
               viewType: FormNodeViewTypes.TIME,
               editType: FormNodeEditTypes.DATETIME,
-              validators: [{ name: ValidatorNames.Required }, { name: ValidatorNames.CopyValueToRootControl, args: 'testEndTimestamp' }]
+              validators: [
+                { name: ValidatorNames.Required },
+                { name: ValidatorNames.AheadOfDate, args: 'testTypeStartTimestamp' },
```
Comment (gjulien-bjss) [inline src/app/forms/templates/test-records/section-templates/test/specialist-test-section-group4.template.ts:131]: Does this one need to be on the `testTypeEndTimeStamp`?

File: src/app/forms/validators/custom-validators.ts [hunk 3]
```diff
@@ -156,6 +156,25 @@ export class CustomValidators {
...
+
+        if (new Date(date) < new Date(siblingValue)) {
+          return { aheadOfDate: { sibling: siblingControl.meta.label } };
+        }
```
Comment (gjulien-bjss) [inline src/app/forms/validators/custom-validators.ts:171]:
```suggestion
        const siblingControl = control.parent.get(sibling) as CustomFormControl;

        if (new Date(control.value) < new Date(siblingControl.value)) {
          return { aheadOfDate: { sibling: siblingControl.meta.label } };
        }
```

**Review by gjulien-bjss — APPROVED** [review]

**Review by Ruairidh-BJSS — APPROVED** [review]

**Pull request merged**